Modify linux build instructions. (#290)
Swartz27/xray-16: doc/howto/build-linux.md

## Build and configure X-Ray 1.6 on Linux

Make sure you are using STALKER Call Of Pripyat v1.6.02 and your system contains the following packages:
git, cmake, GLEW, Freeimage, Lockfile, Openal, TBB, Crypto++, pugixml, Theora, Ogg, SDL2, Lzo, Jpeg.

To install software above in **Debian** and **Ubuntu** derivatives just run the command:

```

> sudo apt install git cmake libglew-dev libfreeimage-dev liblockfile-dev libopenal-dev libtbb-dev libcrypto ++ - dev libpugixml-dev libogg-dev libtheora-dev libvorbis-dev libsdl2-dev liblzo-dev-libjdg2-dev

Reviewer: fix typo
libcrypto++-dev ... liblzo**2**-dev libjdg2-dev

```

In case you are **Fedora** user:
```
sudo dnf install git cmake glew-devel freeimage-devel liblockfile-devel openal-devel tbb-devel cryptopp-devel pugixml-devel libogg-devel libtheora-devel libvorbis-devel SDL2-devel lzo-devel libjpeg-turbo-devel
```
Next clone the repo with its submodules:
```

> git clone https://github.com/OpenXRay/xray-16.git && git submodule update --init --recursive

Reviewer 2: Lil' bit wrong - to download submodules you'll have to cd to cloned xray-16 folder first, or else you'll get error
Should be like that:
git clone https://github.com/OpenXRay/xray-16.git && **cd xray-16 &&** git submodule update --init --recursive

Reviewer 3: ```suggestion
git clone https://github.com/OpenXRay/xray-16.git --recursive
```

```
After the cloning is complete, the xray-16 directory should appear in the directory you invoked command above.
Next it is needed to create directory for binary output and navigate there:
```
cd xray-16 && mkdir bin && cd bin
```
Please pay attention to the errors in next stage. Since they can occur in case of absence of any components.
So, prepare your source to be compiled with:
```
cmake ..
```
You can customize build passing arguments to cmake.
To enable debugging:
```
cmake .. -DCMAKE_BUILD_TYPE = RelWithDebInfo # The string with the debug flag is no longer supported
```
To use clang:
```
CC=clang CXX=clang++ cmake ..
```
To include all optimizations for your machine:
```
CFLAGS="-march=native" CXXFLAGS="-march=native" cmake ..
```
Finally
```
make -jX
```
Where X your processor's number of cores
Also, to output the log and errors to different files:
```

> make -jX > out.log 2 > error.log

Reviewer: fix typo
make -jX >out.log 2>error.log

```
To use linux binaries with the game we need to create directory `bin-linux` there.
So, once compilation is done in the `xray-16/bin` directory execute the command:
```
make DESTDIR=/path/to/your/SCOP/bin-linux install
```
And the `bin-linux` directory will appear where the game is located.

> Yes, `/path/to/your/SCOP` it's the path to your legal copy of STALKER CoP.

Reviewer 3: ```suggestion
Yes, `/path/to/your/SCOP` it's the path to your legal copy of S.T.A.L.K.E.R.: Call of Pripyat,
```

> Backup `res` folder and `fsgame.ltx` in the original game folder in the form you obtained by installing the game from Steam, GOG or license disc

Reviewer 3: There's no `res` folder in original COP
```suggestion
Backup your `fsgame.ltx` in the original game folder in the form you obtained it by installing the game from Steam, GOG or license disc
```

> Next copy from `xray-16` `res` folder and `fsgame.ltx` file to the `/path/to/your/SCOP`.

Reviewer 3: There's no `res` folder in COP, so we should copy the content of this folder, not the folder itself.
```suggestion
Next copy `fsgame.ltx` file and the contents of `res` folder to the `/path/to/your/SCOP`.
```

To run the game invoke one of the following commands:
If you want to debug the game, you need to prevent input capture:
```

> setxkbmap -option grab: break_actions

Reviewer 4: `setxkbmap -option grab:break_actions` - no space between `grab:` and `break_action`

```
Now you can run the game being in `bin-linux` directory using:
```
./xr_3da.sh -fsltx ../fsgame.ltx
```
if you want to debug using gdb:
```

> DEBUGGER = "gdb --ex=r --args" ./xr_3da.sh -fsltx ../fsgame.ltx

Reviewer 4: no spaces around `=`:
`DEBUGGER="gdb --ex=r --args" ./xr_3da.sh -fsltx ../fsgame.ltx`

```

You can also create a "Shortcut" to do this, create a file S.T.A.L.K.E.R.desktop with the following content
```
[Desktop Entry]
Type = Application
Terminal = false
Categories = Game
# The icon should be located at ~/.local/share/icons/
Icon = stalker_cop
# Full path to working directory
Path = /path/to/your/SCOP/linux-bin
# Full path to binaries
Exec = /path/to/your/SCOP/linux-bin/xr_3da.sh -fsltx ../fsgame.ltx

> Name = S.T.A.L.K.E.R. Openxray engine

Reviewer 3: ```suggestion
Name = S.T.A.L.K.E.R.: Call of Pripyat (Powered by OpenXRay)
```

> Comment=S.T.A.L.K.E.R. (OpenXRay Engine)

Reviewer 3: ```suggestion
Comment=S.T.A.L.K.E.R.: Call of Pripyat (Powered by OpenXRay)
```

> Comment[ru_RU]=S.T.A.L.K.E.R. (OpenXRay Engine)

Reviewer 3: ```suggestion
Comment[ru_RU]=S.T.A.L.K.E.R.: Зов Припяти (Powered by OpenXRay)
```

> Comment[uk]=S.T.A.L.K.E.R. (OpenXRay Engine)

Reviewer 3: ```suggestion
Comment[uk]=S.T.A.L.K.E.R.: Поклик Прип'яті (Powered by OpenXRay)
```

> GenericName=Not the official port of the GSC XRay 1.6.02 engine from the OpenXRay team

Reviewer 3: ```suggestion
GenericName=Unofficial X-Ray Engine Linux port by OpenXRay team (Originally developed by GSC Game World)
```

> GenericName[ru_RU]=Не официальный порт движка GSC XRay 1.6.02 от команды OpenXRay

Reviewer 3: ```suggestion
GenericName[ru_RU]=Неофициальный порт игрового движка X-Ray Engine от команды OpenXRay (Оригинальный движок разработан компанией GSC Game World)
```

> GenericName[uk]=Не офіційний порт движка GSC XRay 1.6.02 від команди OpenXRay

Reviewer 3: ```suggestion
GenericName[uk]=Неофіційний порт ігрового движка X-Ray Engine від команди OpenXRay (Оригінальний движок розроблений компанією GSC Game World)
```

Reviewer 5: двигун
